doc/subsystems/shell/shell.rst

```diff
@@ -0,0 +1,271 @@
+:cpp:func:`shell_help_print`. The former is more convinient as it also
+checks for valid arguments count.
+
+
+The following code is an example of how commands can be processed:
+
+.. code-block:: c
+
+	static void cmd_resize(const struct shell *shell, size_t argc,
+				char **argv)
+	{
+		int err;
+
+		if (!IS_ENABLED(CONFIG_SHELL_CMDS_RESIZE)) {
+			shell_fprintf(shell, SHELL_ERROR,
+					"Command not supported.\r\n");
+			return;
+		}
+
+		/* function responsible for printing command help */
+		if (!shell_cmd_precheck(shell, (argc <= 2), NULL, 0)) {
+			return;
+		}
+
+		if (argc != 1) {
+			shell_fprintf(shell, SHELL_ERROR, "%s:%s%s\r\n",
+				      argv[0],
+				      SHELL_MSG_UNKNOWN_PARAMETER, argv[1]);
+			return;
+		}
+
+		err = terminal_size_get(shell);
+		if (err != 0) {
+			shell->ctx->vt100_ctx.cons.terminal_wid =
+					SHELL_DEFAULT_TERMINAL_WIDTH;
+			shell->ctx->vt100_ctx.cons.terminal_hei =
+					SHELL_DEFAULT_TERMINAL_HEIGHT;
+			shell_fprintf(shell, SHELL_WARNING,
+				      "No response from the terminal, assumed"
+					" 80x24 screen size\r\n");
+		}
+	}
+
+When possible, use subcommands instead of options.  Options apply mainly in the
+case when an argument with ``-`` or ``--`` is requested. The main benefit of
+using subcommands is that they can be prompted or completed with the Tab key.
+In addition, subcommands can have their own handler, which limits the usage of
+"if - else if" statements combination with the strcmp function.
+
+.. warning::
+	Do not use function :cpp:func:`shell_fprintf` outside of the command
+	handler because this might lead to incorrect text display on the
+	screen. If any text should be displayed outside of the command context,
+	then use the :ref:`logger`.
+
+Wildcards
+*********
+
+The shell module can handle wildcards. Wildcards are interpreted correctly
+when expanded command and its subcommands do not have a handler. For example,
+if you want to set logging level to ``err`` for the ``app`` and ``app_test``
+modules you can execute the following command:
+
+.. code-block:: none
+
+	log enable err a*
+
+.. image:: images/wildcard.png
+      :align: center
+      :alt: Wildcard usage example
+
+Meta keys
+*********
+
+The shell module supports the following meta keys:
+
+.. list-table:: Implemented meta keys
+   :widths: 10 40
+   :header-rows: 1
+
+   * - Meta keys
+     - Action
+   * - ctrl + a
+     - Moves the cursor to the beginning of the line.
+   * - ctrl + c
+     - Preserves the last command on the screen and starts a new command in a new line.
+   * - ctrl + e
+     - Moves the cursor to the end of the line.
+   * - ctrl + l
+     - Clears the screen and leaves the currently typed command at the top of the screen.
+   * - ctrl + u
+     - Clears the currently typed command.
+   * - ctrl + w
+     - Removes the word or part of the word to the left of the cursor. Words separated by period instead of space are treated as one word.
```

Comment:
for the long line you can write this as:

    * - ctrl + w
      - Removes the word or part of the word to the left of the cursor. 
        Words separated by period instead of space are treated
        as one word.